File: management-domain/Digital-marketing/Backend/Java/budget-management-service/src/main/java/com/gogidix/digitalmarketing/shared/exception/ValidationException.java
package com.gogidix.digitalmarketing.shared.exception;

/**
 * Exception thrown when validation fails.
 */
public class ValidationException extends DomainException {

    public ValidationException(String message) {
        super("VALIDATION_ERROR", message);
    }

    public ValidationException(String field, String reason) {
        super("VALIDATION_ERROR", String.format("Validation failed for field '%s': %s", field, reason));
    }

    public ValidationException(String message, Throwable cause) {
        super("VALIDATION_ERROR", message, cause);
    }
}

File: management-domain/Finance-department/Backend/Nodes/finance-automation-service/invoice-processing-service/src/shared/exceptions/validation.exception.ts
import { HttpStatus } from '@nestjs/common';
import { BaseException } from './base.exception';

/**
 * Exception thrown when validation fails
 */
export class ValidationException extends BaseException {
  constructor(message: string, violations: Array<{ field: string; message: string }>) {
    super(
      message,
      HttpStatus.BAD_REQUEST,
      'VALIDATION_ERROR',
      { violations },
    );
  }
}
